libraries/TGRSIAnalysis/TAngularCorrelation/TAngularCorrelation.cxx

```diff
@@ -0,0 +1,827 @@
+   ///////////////////////////////////////////////////////
+
+   std::vector<TCanvas*> c;
+
+   // loop over the indices
+   for (Int_t i=1;i<=indexmax-indexmin;i++) {
+      Int_t index = hst->GetYaxis()->GetBinLowEdge(i);
+
+      if (visualization) {
+         // find the correct pad
+         Int_t canvas = (i-1)/16;
+         Int_t pad = (i-1)%16;
+         if (pad==0) {
+            TCanvas* temp = new TCanvas(Form("c%i",canvas),Form("c%i",canvas),800,800);
+            temp->Divide(4,4);
+            c.push_back(temp);
+         }
+         // go to canvas pad
+         c[canvas]->cd(pad+1);
+      }
+
+      // pull individual slice
+      TH1D* temphst = hst->ProjectionX(Form("%s_proj%i",hst2dname,index),i,i);
+      temphst->SetStats(0);
+      temphst->SetTitle(Form("%s: angular index %i",hst->GetTitle(),index));
+      this->Set1DSlice(index,temphst);
+
+      // draw the slice to canvas
+      if (visualization) this->Get1DSlice(index)->Draw("pe1");
+
+      // if there are too few counts, continue on (you can fit it manually later)
+      if (temphst->Integral()<100) continue;
+
+      // rename TPeak
+      peak->SetName(Form("%s_proj%i_peak",hst2dname,index));
+
+      // fit TPeak
+      Bool_t fitresult = peak->Fit(temphst,"Q");
+      if (!fitresult) continue; // if fit failed, continue on to next index
+
+      // assign TPeak to fPeaks array
+      this->SetPeak(index,(TPeak*) temphst->GetFunction(Form("%s_proj%i_peak",hst2dname,index)));
+
+      // extract area
+      Double_t area = ((TPeak*) this->GetPeak(index))->GetArea();
+      Double_t area_err = ((TPeak*) this->GetPeak(index))->GetAreaErr();
+
+      // fill histogram with area
+      newhst->SetBinContent(i,area);
+      newhst->SetBinError(i,area_err);
+   }
+
+   ///////////////////////////////////////////////////////
+   // Create diagnostic window
+   ///////////////////////////////////////////////////////
+
+   // initialize canvas
+   TCanvas* c_diag = new TCanvas("c_diag",Form("Diagnostics for fitting %i keV peak",(Int_t)peak->GetCentroid()),800,800);
+   c_diag->Divide(2,2);
+
+   // plot index correlation
+   c_diag->cd(1);
+   newhst->Draw();
+
+   // create plots for chi^2, centroid, and fwhm
+   TH1D* chi2hst = new TH1D(Form("%s_chi2",hst1dname),Form("%s: #chi^{2};Angular index;#chi^{2}/NDF value",newhst->GetTitle()),indexbins,indexmin,indexmax);
+   TH1D* centroidhst = new TH1D(Form("%s_centroid",hst1dname),Form("%s: Peak centroid;Angular index;Peak centroid (keV)",newhst->GetTitle()),indexbins,indexmin,indexmax);
+   TH1D* fwhmhst = new TH1D(Form("%s_fwhm",hst1dname),Form("%s: FWHM;Angular index;FWHM (keV)",newhst->GetTitle()),indexbins,indexmin,indexmax);
+   for (std::map<Int_t,TPeak*>::iterator it=fPeaks.begin(); it!=fPeaks.end(); ++it) {
+      Int_t index = it->first;
+      Int_t bin = newhst->FindBin(index);
+
+      // extract pertinent values from TPeaks
+      Double_t chi2 = ((TPeak*) this->GetPeak(index))->GetChisquare();
```

Comment:
per discussion, `static_cast<TPeak*>(this->GetPeak(index))->GetChisquare()` please.